models/users.js

```diff
@@ -412,6 +414,33 @@ const getUsersBasedOnFilter = async (query) => {
     const filteredUserDocs = userDocs.filter((doc) => !doc.roles?.archived);
     return filteredUserDocs;
   }
+
+  const { role: roleQuery, verified: verifiedQuery } = query;
+
+  if (roleQuery) {
+    const filteredUsers = [];
+    const snapshot = await userModel.where(`roles.${roleQuery}`, "==", true).get();
+    snapshot.forEach((doc) => {
+      filteredUsers.push({
+        id: doc.id,
+        ...doc.data(),
+      });
+    });
+
+    return filteredUsers.filter((user) => !user.roles?.archived);
+  }
+  if (verifiedQuery === "true") {
+    const filteredUsers = [];
+    const snapshot = await userModel.where("discordId", "!=", null).get();
+    snapshot.forEach((doc) => {
+      filteredUsers.push({
+        id: doc.id,
+        ...doc.data(),
+      });
+    });
+
+    return filteredUsers.filter((user) => !user.roles?.archived);
```

Comment:
what if `role` and `verified` both are passed in the query. 🤔

Reply:
According to the code, `role` has a higher precedence over `verified`.

Reply:
or have we missed the case in which both are present.

Reply:
I have not missed the case where both are passed; it just was not the requirement currently and it didn't make sense to increase the code complexity unnecessarily. I had discussed with Ankush that the current search function here is not scalable and requires a lot of if-else statements, so just implementing the requirements here.

I also considered the case here what if status and role both are passed.

Reply:
> According to the code, `role` has a higher precedence over `verified`.

ok so as @DashDeipayan mentioned in that case then `role` gets a higher precedence over `verified`.

Reply:
We can sit and discuss ways of improving it 🙂 if delivering this feature isn't urgent/we have some time